### PR TITLE
Python client: Improve docstrings, generate docs

### PR DESCRIPTION
**Pull request by jayofdoom**
- Improves docstrings and makes them compliant with RST format
  expected by sphinx.
- Adds configuration and automation for generating jekyll-markdown
  files from docstrings (for use with gh-pages)
- Adds documenation on how to use the generation.

The docstrings in the code can still use significant improvement
as our understanding of the API and python types involved improve,
however this is better than what we have today and provides a basis
for future improvement.

First half of changes needed for #1027.

### CODE REVIEW

File: client/python/armada_client/client.py [hunk 1]
```diff
@@ -89,36 +123,72 @@ def reprioritize_jobs(
         return response
 
     def create_queue(self, name: str, **queue_params):
-        """Implementation of CreateQueue rpc function"""
+        """Create the queue by name.
+
+        TODO(JayF) what are queue params?
```
Comment (kannon92) [inline client/python/armada_client/client.py:128]: Its a proto object that represents the Queue object.  

Its a nested object.

https://github.com/G-Research/armada/blob/master/pkg/api/submit.proto#L89

**Review by kannon92 — DISMISSED** [review]

File: client/python/armada_client/client.py (hunk 1 above)
Comment (jayofdoom) [reply client/python/armada_client/client.py:128]: That's the type in general terms, yes, but it's not the python type of the object. That's what I really need to make these strings more accurate with typing. 

I did mean to remove these TODOs though, I'll do that now.

Comment (kannon92) [reply client/python/armada_client/client.py:128]: ACK.

Comment (jayofdoom) [reply client/python/armada_client/client.py]: Fixed via amended commit.

**Review by kannon92 — APPROVED** [review]

**Review by severinson — APPROVED** [review]

**Pull request merged**